lib/common/xml.c

```diff
@@ -1848,47 +1848,63 @@ pcmk__xml2text(const xmlNode *data, uint32_t options, GString *buffer,
     }
 }
 
-char *
-dump_xml_formatted_with_text(const xmlNode *xml)
+/*!
+ * \internal
+ * \brief Dump an XML tree to a string
+ *
+ * \param[in] xml    XML tree to dump
+ * \param[in] flags  Group of <tt>enum pcmk__xml_fmt_options</tt> flags
+ *
+ * \return Newly allocated string representation of \p xml
+ *
+ * \note The caller is responsible for freeing the return value using
+ *       \c g_free().
+ */
+gchar *
+pcmk__xml_dump(const xmlNode *xml, uint32_t flags)
```

Comment:
I think `pcmk__xml2text()` vs `pcmk__xml_dump()` is confusing ... this is simple enough that all callers can use `pcmk__xml2text()` directly

Reply:
That's fine with me. I kept these as wrappers to avoid the callers having to deal with the `GString`. (`gchar` can be used exactly the same as `char` except that it's freed differently). May not be worth the confusion.

Hmm, if we drop `pcmk__xml2text()`, then do you prefer `pcmk__xml_dump()` (libxml2 uses "dump" for this type of thing, and we're sort of paralleling it) or `pcmk__xml_text()` (since that's our usual convention for string representations)? For the latter, the fact that XML text nodes exist adds a little wrinkle.

Reply:
`pcmk__xml_string()` makes sense, especially since we're creating a GString